src/utils/LinkedInPage.js

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { useLinkedIn } from "react-linkedin-login-oauth2";
+import { getAccessTokenData, getMemberDetails } from "./LinkedInUtils";  
+
+const LinkedInPage = () =>{
+
+  // get the user authorization token from linkedin login
+  const{ linkedInLogin } = useLinkedIn({
+    clientId: process.env.REACT_APP_LINKEDIN_CLIENT_ID,
+    redirectUri: `${window.location.origin}/linkedin/callback`,
+    onSuccess: (authCode) => {
+      console.log("AuthCode: ",authCode);
+      setAuthCode(authCode);
+      setErrorMessage("");
+    },
+    scope: ["w_member_social","openid","profile","email"],
+    onError: (error) => {
+      console.log(error);
+      // setCode("");
+      setErrorMessage(error.errorMessage);
+    },
+  });
+
+
+  const [authCode, setAuthCode] = React.useState("");
+  const [errorMessage, setErrorMessage] = React.useState("");
+  let accessTokenData;
+
+
+  if (authCode){
+    // get the user access token from linkedin api
+    const getAccessToken = async () => {
+      try {
+        accessTokenData = await getAccessTokenData(authCode);
+        console.log("accessTokenData: ", accessTokenData);
+        // if (accessTokenData){
+        //   const memberDetails = await getMemberDetails(accessTokenData.data.access_token);
+        //   console.log("Member Details: ", memberDetails);
+        // }
+
+      }catch (error) {
+        console.error("Error fetching access token:", error);
+      }
+    }
+    getAccessToken();
```

Comment:
Are you setting this access token somewhere? Why don't you have an onLoad with a useEffect and set this token in localstorage to start?

In reality the token should not exist in front end code as you don't want tokens and keys to leak out.

Reply:
i have moved this to a utils.js file and the process goes like this: LinkedinLogin->Get authcode->use authcode to get the Accesstoken-> use accesstoken to get the memberId-> use MemberId to post the content to LinkedIn.